src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ImagePreviewer.java

```diff
@@ -522,6 +553,29 @@ private static void onPreviewImage(DocDisplay display,
       panel.setPopupPosition(coordinates.getPageX(), coordinates.getPageY() + 20);
       panel.show();
    }
+   
+   private static final native void injectAttributes(Element el, String attributes)
+   /*-{
+      // create an element, inject inner HTML, and then extract attributes
+      var ctr = $doc.createElement("div");
+      ctr.innerHTML = "<div " + attributes + ">";
```

Comment:
Presuming that these `attributes` are coming from Markdown, we don't have to be too paranoid, because it is already possible to write whatever HTML you want in Markdown. So there's no new HTML injection capability here. :-) So sanitization is probably appropriate, but primarily with the goal of being fault-tolerant to typos or other unintentional malformations. Maybe parse to attr-value pairs, and then add each pair using the DOM rather than innerHTML?